Cython/Utility/Overflow.c

```diff
@@ -95,20 +108,28 @@ static CYTHON_INLINE {{UINT}} __Pyx_mul_{{NAME}}_checking_overflow({{UINT}} a, {
         return r;
 #endif
     } else {
-        {{UINT}} prod = a * b;
-        double dprod = ((double) a) * ((double) b);
+        // somewhat surprisingly, at least on x86_64 it's quicker to use floats (c.f. the const
+        // variant below)
```

Comment:
Ok, but – is the value range large enough to cover an `unsigned long long` multiplication?

Reply:
Looking at this much more closely, this is actually unsafe when `unsigned long long` is 64-bit (the original intent of this file was to allow false-positive overflow reporting, but not false-negatives, but I realise elsewhere we always raise OverflowError if this is truthy) unless we tightly control rounding modes due to 2^64 - 1 not being a possible value (and we probably don't want to get into changing rounding mode because that's a world of pain!), which is likewise true with a double.

I'm also doubting myself as to whether this is actually quicker v. the const version for the unsigned case anyway…